src/util/designator.h

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_UTIL_DESIGNATOR_H
+#define BITCOIN_UTIL_DESIGNATOR_H
+
+/**
+ * Designated initializers can be used to avoid ordering mishaps in aggregate
+ * initialization. However, they do not prevent uninitialized members. The
+ * checks can be disabled by defining DISABLE_DESIGNATED_INITIALIZER_ERRORS.
+ * This should only be needed on MSVC 2019. MSVC 2022 supports them with the
+ * option "/std:c++20"
+ */
+#ifndef DISABLE_DESIGNATED_INITIALIZER_ERRORS
+#define Desig(field_name) .field_name =
```

Comment:
FWIW, I don't particularly like defining our own macro for this. Would have preferred to either use the language feature as-is, or not.

Reply:
Fair enough, no objection to revert it.

Reply:
Do we really need to support MSVC 2019?

Reply:
Cirrus doesn't supprt VS2022 yet https://cirrus-ci.org/guide/windows/.

Reply:
No, that's the point. We should stop supporting it, as it is know to be broken (it can't compile c++20, even though the flag exists). However, no one was successful in bumping the CI task to msvc22.

Maybe a bounty task? :sweat_smile: @jamesob

Reply:
> #define Desig(field_name, value) .field_name = (value)

see previous discussion: https://github.com/bitcoin/bitcoin/pull/24531#discussion_r885998944

Reply:
#25472 :tiger2: